src/tasks/base-task.ts

```diff
@@ -0,0 +1,51 @@
+// global
+import { FastifyLoggerInstance } from 'fastify';
+import {
+  Actor,
+  DatabaseTransactionHandler,
+  IndividualResultType,
+  PostHookHandlerType,
+  PreHookHandlerType,
+  Task,
+  TaskStatus,
+} from 'graasp';
+
+export abstract class BaseTask<A extends Actor, R> implements Task<A, R> {
+  protected _result: R;
+  protected _message: string;
+  readonly actor: A;
+  protected _partialSubtasks: boolean;
+
+  status: TaskStatus;
+  targetId: string;
+  data: Partial<IndividualResultType<R>>;
+  preHookHandler?: PreHookHandlerType<R>;
+  postHookHandler?: PostHookHandlerType<R>;
+
+  getInput?: () => unknown;
+  getResult?: () => unknown;
+
+  constructor(actor: A) {
+    this.actor = actor;
+    this.status = 'NEW';
```

Comment:
Why don't we have these strings factored out again?

Reply:
These are not factored out in @graasp/types yet so we cannot match the types. At some point it will be unified :/